runtime-sdk/src/context.rs

```diff
@@ -83,16 +92,43 @@ pub trait Context {
         self.mode() == Mode::SimulateTx
     }
 
-    /// Whether expensive queries are allowed based on local configuration.
-    ///
-    /// This method will always return `true` in `Mode::ExecuteTx` contexts.
-    fn are_expensive_queries_allowed(&self) -> bool {
-        if self.mode() == Mode::ExecuteTx {
-            return true;
+    /// Whether smart contracts should be executed in this context.
+    fn should_execute_contracts(&self) -> bool {
+        match self.mode() {
+            // When actually executing a transaction, we always run contracts.
+            Mode::ExecuteTx => true,
+            Mode::SimulateTx => self
+                .local_config(LOCAL_CONFIG_ESTIMATE_GAS_BY_SIMULATING_CONTRACTS)
+                .unwrap_or_default(),
+            // When just checking a transaction, we always want to be fast and skip contracts.
+            Mode::CheckTx => false,
         }
+    }
 
-        self.local_config(LOCAL_CONFIG_ALLOW_EXPENSIVE_QUERIES)
```

Comment:
On second thought it may be good if we maintained backwards compatibility and use a similar default in case the older config option is present.

Reply:
:+1: done